monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py

```diff
@@ -0,0 +1,54 @@
+from jsonschema import validate
+
+from common.agent_configuration import AgentConfiguration
+from monkey_island.cc.repositories import (
+    IAgentConfigurationRepository,
+    PluginConfigurationValidationError,
+    RetrievalError,
+)
+from monkey_island.cc.repositories.utils import AgentConfigurationSchemaCompiler
+
+
+class AgentConfigurationValidationDecorator(IAgentConfigurationRepository):
+    """
+    A IAgentConfigurationRepository decorator that validates the agent configuration against
+    the jsonschema describing it. Decorator makes sure that only valid configurations get
+    pulled/pushed to repositories.
+    """
```

Comment:
This format is incorrect. It should be:

```
"""
Single-line summary

Multi-line description
"""
```

See https://peps.python.org/pep-0257/#multi-line-docstrings for more details.